src/subtitles/parser.rs

```diff
@@ -0,0 +1,205 @@
+use super::types::SubtitleEntry;
+use super::utils::format_timestamp;
+use crate::errors::MediaParserResult;
+#[cfg(test)]
+use std::io;
+
+/// Parse subtitle sample data based on codec type
+pub fn parse_subtitle_sample_data(
+    data: &[u8],
+    timestamp: f64,
+    codec_type: &str,
+) -> MediaParserResult<Vec<SubtitleEntry>> {
+    if data.is_empty() {
+        return Ok(Vec::new());
+    }
+
+    match codec_type {
+        "tx3g" => parse_tx3g_subtitle(data, timestamp),
+        "wvtt" => parse_webvtt_subtitle(data, timestamp),
+        "stpp" => parse_ttml_subtitle(data, timestamp),
+        "sbtl" | "subt" => parse_generic_subtitle(data, timestamp),
+        _ => {
+            println!(
+                "Unknown subtitle codec: {}, trying generic parser",
+                codec_type
+            );
+            parse_generic_subtitle(data, timestamp)
+        }
+    }
+}
+
+/// Parse TX3G (3GPP Timed Text) subtitle format
+fn parse_tx3g_subtitle(data: &[u8], timestamp: f64) -> MediaParserResult<Vec<SubtitleEntry>> {
+    if data.len() < 2 {
+        return Ok(Vec::new());
+    }
+
+    // TX3G format: 2-byte text length + text data
+    let text_length = u16::from_be_bytes([data[0], data[1]]) as usize;
+
+    if text_length == 0 || data.len() < 2 + text_length {
+        return Ok(Vec::new());
+    }
+
+    let text_data = &data[2..2 + text_length];
+
+    // Try to decode as UTF-8
+    if let Ok(text) = String::from_utf8(text_data.to_vec()) {
+        if !text.trim().is_empty() {
+            return Ok(vec![SubtitleEntry {
+                start: format_timestamp(timestamp),
+                end: format_timestamp(timestamp + 2.0), // Default 2-second duration
```

Comment:
I may be misunderstanding what's happening, but I don't think that assuming that all subtitles have 2 second duration is what we want. We should be grabbing the real end times specified in the subtitle data, right?

This doesn't seem to be a default as much as it is just overriding all of the end timestamp data.